### ISSUE TITLE
Style Guide - Provide guidance/examples on how to use in unit tests

### ISSUE THREAD

**Issue by jonboiser** [learningequality/kolibri#2829]
### Observed behavior

In my experience, interacting with Keen-UI components (e.g. clicking, changing inputs, etc.) has forced me to use a different approach than with plain HTML elements. For example, to cause a behavior on an HTML element, I can simply do `el.trigger('click')`, whereas with the Vue component, I will sometimes need to reach into the component's view model and simulate an Event by calling a view model method directly (e.g. `component.vm.handleClickYes()`).

Whether this is correct from my (i.e. the test writer) perspective or the component author is up for debate, but it would be nice if there were code examples for each style guide element explaining how one would do basic testing if I were to use the component in the app.

### Expected behavior

Some suggestions for the style-guide

* **Interactivity** explain how to click/change/etc. the form element
* **assertions** how to make assertions on relevant part of the component or the bigger part of the app

**Comment by indirectlylit** [comment]
agree this should be documented, but would advocate putting it in the dev docs rather than the style guide, because it won't be very meaningful for non-devs.

We have a location here which is appropriate: https://kolibri-dev.readthedocs.io/en/develop/frontend_architecture/unit_testing.html

**Comment by jonboiser** [comment]
The style guide already has a lot of code-specific information, like the component API. At its inception, it seems like it was more of API documentation for K* components, but with some extra text describing "style" issues. If we really want to keep the dev-stuff out, maybe the style and API content should be separated.

For me, it would be a little bit of extra work to 

1. Use the style guide to get some idea of how to use a component
1. Go to a different page to see how to test it

Also, since the style guide documents the component's API and events, etc., it's already dev-focused. And, for example, if you change an event name for a component, then the testing docs will now be stale (if they depend on that event name).

**Comment by indirectlylit** [comment]
ah, I may have misunderstood the intention.

Are you envisioning that there would be some per-component guidance for testing? I.e., something specific to textboxes and something else specific to buttons?

**Comment by jonboiser** [comment]
Yeah, something like "How to trigger" a change in KCheckbox or in KTextbox should be in the appropriate docs page for those elements.

**Comment by rtibbles** [comment]
I think this is now fairly well covered by recent updates to our documentation, especially the Vue Testing Library template here: https://kolibri-dev.readthedocs.io/en/develop/frontend_architecture/testing_layout.html
